modules/ismp/pallet/pallet-asset-transfer/src/lib.rs

```diff
@@ -0,0 +1,477 @@
+        pub fn set_params(
+            origin: OriginFor<T>,
+            update: TokenGatewayParamsUpdate,
+        ) -> DispatchResult {
+            <T as pallet_ismp::Config>::AdminOrigin::ensure_origin(origin)?;
+            let mut current_params = Params::<T>::get().unwrap_or(T::Params::get());
+
+            if let Some(address) = update.token_gateway_address {
+                current_params.token_gateway_address = address;
+            }
+
+            if let Some(protocol_fee_percentage) = update.protocol_fee_percentage {
+                current_params.protocol_fee_percentage = protocol_fee_percentage;
+            }
+
+            if let Some(dot_asset_id) = update.dot_asset_id {
+                current_params.dot_asset_id = dot_asset_id;
+            }
+
+            Params::<T>::put(current_params);
+            Ok(())
+        }
+    }
+}
+
+impl<T: Config> Pallet<T>
+where
+    u128: From<<T::Assets as fungibles::Inspect<T::AccountId>>::Balance>,
+    T::AccountId: Into<[u8; 32]>,
+{
+    pub fn account_id() -> T::AccountId {
+        T::PalletId::get().into_account_truncating()
+    }
+
+    pub fn protocol_account_id() -> T::AccountId {
+        T::ProtocolAccount::get().into_account_truncating()
+    }
+
+    pub fn token_gateway_address() -> H160 {
+        Params::<T>::get().unwrap_or(<T as Config>::Params::get()).token_gateway_address
+    }
+
+    pub fn protocol_fee_percentage() -> Percent {
+        Params::<T>::get()
+            .unwrap_or(<T as Config>::Params::get())
+            .protocol_fee_percentage
+    }
+
+    pub fn dot_asset_id() -> H256 {
+        Params::<T>::get().unwrap_or(<T as Config>::Params::get()).dot_asset_id
+    }
+
+    /// Dispatch ismp request to token gateway on destination chain
+    pub fn dispatch_request(
+        multi_account: MultiAccount<T::AccountId>,
+        amount: <T::Assets as fungibles::Inspect<T::AccountId>>::Balance,
+    ) -> Result<(), Error<T>> {
+        let dispatcher = Dispatcher::<T>::default();
+
+        let mut to = [0u8; 32];
+        to[..20].copy_from_slice(&multi_account.evm_account.0);
+        let from: [u8; 32] = multi_account.substrate_account.clone().into();
+        let asset_id = Self::dot_asset_id().0.into();
+        let body = Body {
+            amount: {
+                let amount: u128 = amount.into();
+                let mut bytes = [0u8; 32];
+                U256::from(amount).to_big_endian(&mut bytes);
+                alloy_primitives::U256::from_be_bytes(bytes)
+            },
+            asset_id,
+            redeem: false,
+            from: from.into(),
+            to: to.into(),
+        };
+
+        let dispatch_post = DispatchPost {
+            dest: multi_account.dest_state_machine,
+            from: Self::token_gateway_address().0.to_vec(),
+            to: Self::token_gateway_address().0.to_vec(),
+            timeout_timestamp: multi_account.timeout,
+            data: alloy_rlp::encode(body),
```

Comment:
This should be prefixed with a zero-byte for the handle incoming asset variant.

https://github.com/polytope-labs/hyperbridge/blob/main/evm/src/modules/TokenGateway.sol#L226